diff --git a/tools/buildscripts/qtwebengine_utils.py b/tools/buildscripts/qtwebengine_utils.py
index 75ee076..b24358f 100755
--- a/tools/buildscripts/qtwebengine_utils.py
+++ b/tools/buildscripts/qtwebengine_utils.py
@@ -47,9 +47,12 @@ import sys
 
 def getChromiumSrcDir():
 
+  saved_cwd = os.getcwd()
   qtwebengine_root = os.path.abspath(os.path.join(os.path.dirname(__file__), "../.."))
 
+  os.chdir(qtwebengine_root)
   chrome_src = subprocess.check_output("git config qtwebengine.chromiumsrcdir || true", shell=True).strip()
+  os.chdir(saved_cwd)
   if chrome_src:
     chrome_src = os.path.join(qtwebengine_root, chrome_src)
   if not chrome_src or not os.path.isdir(chrome_src):
